visualization/dashboard/src/MLIDashboard/Controls/GlobalViolinPlot.tsx

```diff
@@ -0,0 +1,116 @@
+        const plotlyProps = _.cloneDeep(GlobalViolinPlot.BasePlotlyProps);
+
+        _.set(plotlyProps, 'layout.xaxis.ticktext', sortVector.map(i => metadata.featureNamesAbridged[i]));
+        _.set(plotlyProps, 'layout.xaxis.tickvals', sortVector.map((val, index) => index));
+
+        const x = [];
+        const y = [];
+        const baseData = plotlyProps.data[0];
+        const dataArray: IData[] = [];
+        sortVector.forEach((featureIndex, xIndex) => {
+            const featureImportance = cohort.unwrap(JointDataset.ReducedLocalImportanceRoot + featureIndex.toString());
+            const data = {...baseData};
+            data.x = new Array(featureImportance.length).fill(xIndex);
+            data.y = featureImportance;
+            data.line = {color: FabricStyles.plotlyColorPalette[0]} as any;
+            dataArray.push(data);
+        });
+        plotlyProps.data = dataArray;
+        return plotlyProps;
+    }, _.isEqual);
+
+
+    private static BasePlotlyProps: IPlotlyProperty = {
+        config: { displaylogo: false, responsive: true, displayModeBar: false  } as any,
+        data: [
+            {
+                type: 'violin',
+                scalemode: 'count',
+                spanmode: 'hard',
+            }
+        ] as any,
+        layout: {
+            dragmode: false,
+            autosize: true,
+            font: {
+                size: 10
+            },
+            hovermode: 'closest',
+            margin: {
+                t: 10, b: 30, r: 210
+            },
+            showlegend: false,
+            yaxis: {
+                automargin: true,
+                title: localization.featureImportance
+            },
+            xaxis: {
+                automargin: true
+            }
+        } as any
+    };
+
+    constructor(props: IGlobalViolinPlotProps) {
+        super(props);
+
+        this.state = {
+            plotlyProps: undefined
+        };
+    }
+
+    public componentDidUpdate(prevProps: IGlobalViolinPlotProps) {
+        if (this.props.sortVector !== prevProps.sortVector) {
+            this.setState({plotlyProps: undefined});
+        }
+    }
+
+    public render(): React.ReactNode {
+        if (this.state.plotlyProps === undefined) {
+            const plotlyProps = GlobalViolinPlot.buildPlotlyProps(this.props.jointDataset, this.props.metadata, this.props.cohort, this.props.sortVector, undefined);
+            this.setState({plotlyProps});
+            return <LoadingSpinner/>;
+        }
+        const minK = Math.min(4, this.props.jointDataset.localExplanationFeatureCount);
+        const maxK = Math.min(30, this.props.jointDataset.localExplanationFeatureCount);
+        const maxStartingK = Math.max(0, this.props.jointDataset.localExplanationFeatureCount - this.props.topK);
+        const relayoutArg = {'xaxis.range': [this.props.startingK - 0.5, this.props.startingK + this.props.topK - 0.5]};
```

Comment:
some of this code looks similar to above, can it be moved to some common component